tests/integration/test_macaroon_auth.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import os
+
+import macaroonbakery.bakery as bakery
+import macaroonbakery.httpbakery as httpbakery
+import macaroonbakery.httpbakery.agent as agent
+from juju.client import client
+from juju.errors import JujuAPIError
+from juju.model import Model
+
+import pytest
+
+from .. import base
+
+log = logging.getLogger(__name__)
+
+
+@base.bootstrapped
+@pytest.mark.asyncio
+async def test_macaroon_auth(event_loop):
+    try:
+       auth_info, username = agent_auth_info()
+    except NoAgentInfo:
+       return
```

Comment:
It would be better for these to skip using [`pytest.skip(reason)`](https://docs.pytest.org/en/latest/skipping.html) to make it clear that they didn't actually succeed.